client/components/domains/connect-domain-step/index.jsx

```diff
@@ -0,0 +1,217 @@
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { domainManagementList } from 'calypso/my-sites/domains/paths';
+import { isMappingVerificationSuccess } from './connect-domain-step-verification-status-parsing.js';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+function ConnectDomainStep( { domain, selectedSite, initialSetupInfo, initialStep, showErrors } ) {
+	const [ mode, setMode ] = useState( modeType.SUGGESTED );
+	const [ step, setStep ] = useState( stepType.START );
+	const [ currentPageSlug, setCurrentPageSlug ] = useState( stepSlug.SUGGESTED_START );
+	const [ progressStepList, setProgressStepList ] = useState( {} );
+	const [ verificationStatus, setVerificationStatus ] = useState( {} );
+	const [ verificationInProgress, setVerificationInProgress ] = useState( false );
+	const [ domainSetupInfo, setDomainSetupInfo ] = useState( defaultDomainSetupInfo );
+	const [ domainSetupInfoError, setDomainSetupInfoError ] = useState( {} );
+	const [ loadingDomainSetupInfo, setLoadingDomainSetupInfo ] = useState( false );
+	const [ stepsDefinition, setStepsDefinition ] = useState( defaultStepsDefinition );
+
+	const baseClassName = 'connect-domain-step';
+	const StepsComponent = stepsDefinition?.[ currentPageSlug ].component;
+	const isStepStart = stepType.START === step;
+
+	const statusRef = useRef( {} );
+
+	const setPage = useCallback(
+		( pageStepSlug ) => {
+			setCurrentPageSlug( pageStepSlug );
+			setStep( stepsDefinition[ pageStepSlug ].step );
+			setMode( stepsDefinition[ pageStepSlug ].mode );
+		},
+		[ stepsDefinition ]
+	);
+
+	const setNextStep = () => {
+		const next = stepsDefinition[ currentPageSlug ]?.next;
+		next && setPage( next );
+	};
+
+	const switchToSuggestedSetup = () => {
+		setPage( stepSlug.SUGGESTED_START );
+	};
+
+	const switchToAdvancedSetup = () => {
+		setPage( stepSlug.ADVANCED_START );
+	};
+
+	useEffect( () => {
+		setStepsDefinition( getStepsDefinition( selectedSite, domain, domainSetupInfo ) );
+	}, [ domain, domainSetupInfo, selectedSite ] );
+
+	useEffect( () => {
+		setCurrentPageSlug( getPageSlug( mode, step, stepsDefinition ) );
+	}, [ mode, step, stepsDefinition ] );
+
+	useEffect( () => {
+		if ( initialStep && Object.values( stepSlug ).includes( initialStep ) ) {
+			setPage( initialStep );
+		}
+	}, [ initialStep, setPage ] );
+
+	useEffect( () => {
+		setProgressStepList( getProgressStepList( mode, stepsDefinition ) );
+	}, [ mode, stepsDefinition ] );
+
+	const verifyConnection = useCallback(
+		( setStepAfterVerify = true ) => {
+			setVerificationStatus( {} );
+			setVerificationInProgress( true );
+			wp.undocumented()
+				.getMappingStatus( domain )
+				.then( ( data ) => {
+					setVerificationStatus( { data } );
+					if ( setStepAfterVerify ) {
+						if ( isMappingVerificationSuccess( mode, data ) ) {
+							setStep( stepType.CONNECTED );
+						} else {
+							setStep( stepType.VERIFYING );
+						}
```

Comment:
Maybe a ternary operator here can improve readability?
```suggestion
						setStep( isMappingVerificationSuccess( mode, data )? stepType.CONNECTED : stepType.VERIFYING );
```

Reply:
It might just be me, but I feel like the ternary has a higher cognitive load there.  And I also tried using `const nextStep = ...` with a ternary and then calling `setStep( nextStep )`, but I think that I like the clear if/else in this case.